Automattic/woocommerce-payments
# BNPL - Fix error adding variable product to cart in some themes

Fixes #6819 

#### Changes proposed in this Pull Request

* Prevent the error in some themes, like Flatsome, that blocks adding of variable product to cart by targeting and reading the quantity div correctly.
* Prevent the BNPL Payment Messaging element from rendering when BNPL payment methods are turned off or not available. 

#### Testing instructions

* Check out this branch on WCPay client
* Make sure you have Afterpay and Affirm are both disabled on the test site, to begin with.
* Install Flatsome theme on your test site
* Browse to the product page of a variable product
* Select some variation and try adding it to the cart
* Try with a different quantity too
* In each of the case you should not see any errors - neither popups, nor any errors in the browser console
* Within the browser console, browse the HTML code of the page. You **should not** see a div with id `payment-method-message`
* Repeat the steps with Affirm and Afterpay enabled
* As expected, you should see relevant offers depending on what product variation and quantity you select

##### Optional testing tip 

This is a tip to visually see if the messaging element is hooked, and hence if `payment-method-message` is rendered. 

Within `WP Admin > Appearance > Customize >> Additional CSS` add this CSS snippet: 

```
#payment-method-message {
    background-color: yellow;
    min-height: 30px;
}
```
Make sure you remove this snippet after testing!

If you checkout to a different branch including `develop` at the moment, and keep BNPL methods disabled, you will see a yellow patch indicating that the messaging hook is active.  
<img width="1286" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/4162931/096d32b7-5305-4357-8acd-71d69fe7d8f1">


If you checkout this branch, you will not see it, since this branch prevents the messaging hook from getting inserted if there are no BNPL methods enabled.  
<img width="1245" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/4162931/34a9cfea-1c31-4654-9c27-86b30d381771">


-------------------


**Post merge**
